src/Microsoft.Health.Dicom.Blob/Features/Storage/BlobMetadataStore.cs

```diff
@@ -106,6 +111,12 @@ public class BlobMetadataStore : IMetadataStore
         }
         catch (Exception ex)
         {
+            if (ex is NotSupportedException)
+            {
+                _telemetryClient
```

Comment:
Interesting to see metric client used here. Can we just use the logger? If you want to track the original exceptions. I did not get a lot of details in the description. Can you add some description for the PR.

Reply:
There are several advantages to recording a metric here instead of a log:
- We can track trends over a longer time range than our log retention period
- We can ensure no private data is collected

In order to collect metrics with Application Insights, this is the [documented method](https://learn.microsoft.com/en-us/azure/azure-monitor/app/get-metric). When we move to an Otel implementation this will need to change, but we are currently collecting metrics this way elsewhere.